code/game/objects/structures/door_assembly.dm

```diff
@@ -124,12 +124,12 @@
 
 /obj/structure/door_assembly/attackby(obj/item/W, mob/user)
 
-	if(IS_PEN(W))
+	if(IS_PEN(W) && CanPhysicallyInteractWith(user, src))
 		var/t = sanitize_safe(input(user, "Enter the name for the door.", src.name, src.created_name), MAX_NAME_LEN)
-		if((!t) || (!in_range(src, usr) && src.loc != usr))
```

Comment:
Need to readd a `CanInteract` or something post-input().

Reply:
Yeah, I see it now. Gonna have to take a look at the other ones too.